Source/JavaScriptCore/runtime/JSArray.h

```diff
@@ -349,7 +355,7 @@ ALWAYS_INLINE void copyArrayElements(T* buffer, unsigned offset, U* source, unsi
 
     if constexpr (std::is_same_v<T, U>) {
         if constexpr (fillMode == ArrayFillMode::Empty) {
-            if constexpr (std::is_same_v<T, double>)
+            if constexpr (std::is_same_v<T, double> || needsGCSafeOps == NeedsGCSafeOps::No)
                 memcpy(buffer + offset, source, sizeof(double) * sourceSize);
```

Comment:
I think the contents of buffer now can be non-double value. But it is okay to leave it as `sizeof(double)`?
( In my knowledge, JSValue and double are the same size, but just in case. )

Reply:
We can add `static_assert` around size.